### PR TITLE
ON-40991 # `ADDRESS_PROPERTY` to conditional predicates schema

### PR DESCRIPTION
**Pull request by divporter**

### CODE REVIEW

File: src/lib/forms-validation/index.ts [hunk 1]
```diff
@@ -142,6 +146,106 @@ function validateFormElementReferences(formElements: FormTypes.FormElement[]) {
...
+          }
+        }
+      }
+      break
+    }
+
+    case 'REPEATABLESET': {
+      if (referencedElement.type !== 'repeatableSet') {
+        throw new Error(
+          `Referenced ${referencedElement.type} type not a repeatableSet}`,
```
Comment (jdawg093) [inline src/lib/forms-validation/index.ts:208]:
```suggestion
          `Referenced ${referencedElement.type} type not a repeatableSet`,
```

File: src/lib/forms-validation/index.ts [hunk 2]
```diff
@@ -142,6 +146,106 @@ function validateFormElementReferences(formElements: FormTypes.FormElement[]) {
...
+const validAddressReferenceTypes: FormTypes.FormElement['type'][] = [
+  'pointAddress',
+  'geoscapeAddress',
+]
+const validNumericReferenceTypes: FormTypes.FormElement['type'][] = [
+  'number',
+  'calculation',
+]
+
+function validateConditionallyShowPredicateReferencedElement({
```
Comment (jdawg093) [inline src/lib/forms-validation/index.ts:158]: Not checking for the `FORM` case?

File: src/lib/forms-validation/index.ts [hunk 3]
```diff
@@ -142,6 +146,106 @@ function validateFormElementReferences(formElements: FormTypes.FormElement[]) {
...
+}
+
+function validateConditionallyShowPredicateReferences(
+  formElements: FormTypes.FormElement[],
+) {
+  const availableFormElements = stripLayoutFormElements(formElements)
+  // Element References
+  for (const element of availableFormElements) {
+    if (element.conditionallyShow) {
+      console.log(element.conditionallyShowPredicates)
```
Comment (jdawg093) [inline src/lib/forms-validation/index.ts:233]: Is this leftover from testing or would you want to keep this here?

File: src/lib/forms-schema/property-schemas.ts [hunk 4]
```diff
@@ -199,6 +206,26 @@ const ConditionalPredicatesItemBaseSchema = Joi.object().keys({
...
+        property: Joi.string().valid('IS_PO_BOX_ADDRESS', 'STATE_EQUALITY'),
+        value: Joi.when('property', {
+          switch: [
+            {
+              is: Joi.valid('IS_PO_BOX_ADDRESS'),
+              then: Joi.boolean().required(),
+            },
+            {
+              is: Joi.valid('STATE_EQUALITY'),
+              then: Joi.string().required(),
```
Comment (mymattcarroll) [inline src/lib/forms-schema/property-schemas.ts:222]: As discussed, lets change this to a typed enum.

File: src/lib/forms-validation/index.ts (hunk 2 above)
Comment (divporter) [reply src/lib/forms-validation/index.ts:158]: It's just an Id, I think the API does, but there is no cross definition reference there for the sdk to use

Comment (jdawg093) [reply src/lib/forms-validation/index.ts]: Forgot about that lil tidbit on the end there 👍

**Pull request merged**